mypy/copytype.py

```diff
@@ -0,0 +1,106 @@
+from typing import Any, cast
+
+from mypy.types import (
+    ProperType, UnboundType, AnyType, NoneType, UninhabitedType, ErasedType, DeletedType,
+    Instance, TypeVarType, ParamSpecType, PartialType, CallableType, TupleType, TypedDictType,
+    LiteralType, UnionType, Overloaded, TypeType, TypeAliasType, UnpackType, Parameters
+)
+from mypy.type_visitor import TypeVisitor
+
+
+def copy_type(t: ProperType) -> ProperType:
+    """Create a shallow copy of a type.
+
+    This can be used to mutate the copy with truthiness information.
+
+    Classes compiled with mypyc don't support copy.copy(), so we need
```

Comment:
Not supporting `copy.copy` is pretty rough. This will also mean that we don't support `pickle`, right?

I don't think that we can reasonably break those without providing an opt-opt.

Reply:
That's a good point. I was planning to provide a way to do this later, but I can add it to this PR. We could have a per-class flag saying that this class (and any subclasses, probably) is picklable, and no attribute would be treated as always defined in the class, and thus we can always provide a way of creating an uninitialized instance.

Reply:
Now it's supported by using `@mypyc_attr(serializable=True)`.